sql-plugin/src/main/311until330-all/scala/com/nvidia/spark/rapids/shims/GpuTypeShims.scala

```diff
@@ -82,15 +83,55 @@ object GpuTypeShims {
     throw new RuntimeException(s"Not support type $dt.")
 
   /**
-   * Spark supports interval type from 320; Spark supports to/from Parquet from 330,
-   * And lots of interval operators from 330, so just return false for Spark330-
+   * Alias supported types for this shim
    */
-  def isDayTimeIntervalType(dt: DataType) : Boolean = false
+  def supportedTypesForAlias: TypeSig = TypeSig.none
 
   /**
-   * Spark supports interval type from 320; Spark supports to/from Parquet from 330,
-   * And lots of interval operators from 330, so just return false for Spark330-
+   * Abs supported types for this shim
    */
-  def isYearMonthIntervalType(dt: DataType) : Boolean = false
+  def supportedTypesForAbs: TypeSig = TypeSig.none
+
+  def isDayTimeTypeAndAbsSupports(dt: DataType): Boolean = false
```

Comment:
Would it be sufficient to have predicate methods like:
```
def isSupportedDayTimeType(dt: DataType): Boolean

def isSupportedYearMonthType(dt: DataType): Boolean
```

We could also have methods that return a TypeSig for additional interval types _if_ they are supported on that Spark version, e.g.:
```
// when intervals are not supported
def supportedIntervalTypes: TypeSig = TypeSig.none
[...]
// when intervals are supported
def supportedIntervalTypes: TypeSig = TypeSig.ansiIntervals
```